uboot/secure_boot/imx8mm_habv4.rst

```diff
@@ -0,0 +1,594 @@
+        spl hab block:         0x7e0fc0 0x0 0x27c00
+
+        Second Loader IMAGE:
+        sld_header_off         0x57c00
+        sld_csf_off            0x58c20
+        sld hab block:         0x401fcdc0 0x57c00 0x1020
+
+
+Creating the CSF description files
+----------------------------------
+
+The CSF description file sample is provided at u-boot/doc/imx/habv4/csf_examples
+
+There are two files since we have two stage bootloader, csf_spl.txt and csf_fit.txt. Create a directory
+csf and copy both the files from u-boot/doc/imx/habv4/csf_examples.
+
+For csf_spl.txt, in the Authenticate Data section replace Blocks with spl hab block content above:
+
+::
+
+        Blocks = 0x7e0fc0 0x0 0x27c00 "flash.bin"
+
+Adjust necessary paths.
+
+For csf_fit.txt, in the Authenticate Data section replace the first line in Blocks with sld hab block content above
+and the next three lines by the output provided by the below command:
+
+::
+
+        imx-mkimage$ make SOC=iMX8MM print_fit_hab
+        ./../scripts/dtb_check.sh imx8mm-ctouch.dtb ctouch.dtb
+        Use u-boot DTB: imx8mm-ctouch.dtb
+        ./../scripts/pad_image.sh tee.bin
+        Pad file tee.bin NOT found
+        ./../scripts/pad_image.sh bl31.bin
+        ./../scripts/pad_image.sh u-boot-nodtb.bin ctouch.dtb
+        u-boot-nodtb.bin + ctouch.dtb are padded to 610720
+        TEE_LOAD_ADDR=0xbe000000 ATF_LOAD_ADDR=0x00920000 VERSION=v1 ./print_fit_hab.sh 0x60000 ctouch.dtb
+        0x40200000 0x5AC00 0x8DCB0 
+        0x4028DCB0 0xE88B0 0x74F0
+        0x920000 0xEFDA0 0x9160
+
+The resulting Authenticate Data section in csf_fit.txt should be:
+
+::
+
+        Blocks = 0x401fcdc0 0x57c00 0x01020 "flash.bin", \
+                     0x40200000 0x5AC00 0x8DCB0 "flash.bin", \
+                     0x4028DCB0 0xE88B0 0x074F0 "flash.bin", \
+                     0x00920000 0xEFDA0 0x09160 "flash.bin"
+
+Adjust necessary paths in both csf_spl.txt and csf_fit.txt.
+
+Signing and Asembling the flash.bin binary
+-------------------------------------------
+
+::
+
+         csf$ ../cst-3.3.1/linux64/bin/cst -i csf_spl.txt -o csf_spl.bin
+         Install SRK
+         Install CSFK
+         Authenticate CSF
+         Install key
+         Authenticate data
+         CSF Processed successfully and signed data available in csf_spl.bin
+
+         csf$ ../cst-3.3.1/linux64/bin/cst -i csf_fit.txt -o csf_fit.bin
+         Install SRK
+         Install CSFK
+         Authenticate CSF
+         Install key
+         Authenticate data
+         CSF Processed successfully and signed data available in csf_fit.bin
+
+         csf$ copy flash.bin from imx-mkimage/IMX8M to csf directory         
+         csf$ cp flash.bin signed_flash.bin
+
+         csf$ dd if=csf_spl.bin of=signed_flash.bin seek=$((0x27c00)) bs=1 conv=notrunc
+         the seek above is csf_off
+
+         csf$ dd if=csf_fit.bin of=signed_flash.bin seek=$((0x58c20)) bs=1 conv=notrunc
+         the seek above is sld_csf_off
+
```

Comment:
Can you create a diagram layout for how signed images get offset? like 
https://wiki.amarulasolutions.com/uboot/secure_boot/imx6_habv4.html#signed-image-format-u-boot